setup.py

```diff
@@ -23,13 +23,20 @@
 from distutils.command.build_ext import build_ext as _build_ext
 
 extra_compile_args = {
-    'msvc': ['/EHsc'],
-    'unix': ['-std=c++11'],
+    'msvc': ['/std:c++17', '/EHsc'],
+    'unix': [
+        '-std=c++17', 
+        '-Wall', 
```

Comment:
let's not do these here. The Python build generated a bunch of warnings regardless